android/src/main/java/com/reactnativea11y/modules/services/KeyboardService.java

```diff
@@ -0,0 +1,92 @@
+package com.reactnativea11y.modules.services;
+
+import static android.content.res.Configuration.HARDKEYBOARDHIDDEN_NO;
+import static android.content.res.Configuration.KEYBOARD_NOKEYS;
+import static android.content.res.Configuration.KEYBOARD_UNDEFINED;
+
+import android.app.Activity;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.res.Configuration;
+import android.util.Log;
+import android.view.View;
+
+import com.facebook.react.bridge.LifecycleEventListener;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.uimanager.IllegalViewOperationException;
+import com.facebook.react.uimanager.UIManagerModule;
+
+public class KeyboardService implements LifecycleEventListener {
+  private final String NEW_CONFIG = "newConfig";
+  private final String ON_CONFIGURATION_CHANGED = "onConfigurationChanged";
+
+  private ReactApplicationContext context;
+  private BroadcastReceiver receiver;
+
+  public boolean isKeyboardConnected() {
+    int keyboard = context.getResources().getConfiguration().keyboard;
+    return keyboard != KEYBOARD_UNDEFINED && keyboard != KEYBOARD_NOKEYS;
+  }
+
+  public KeyboardService(ReactApplicationContext context) {
+    this.context = context;
+    context.addLifecycleEventListener(this);
+    receiver = new BroadcastReceiver() {
+      @Override
+      public void onReceive(Context context, Intent intent) {
+        Configuration newConfig = intent.getParcelableExtra(NEW_CONFIG);
+        keyboardChanged(newConfig.hardKeyboardHidden == HARDKEYBOARDHIDDEN_NO);
+      }
+    };
+    keyboardChanged(isKeyboardConnected());
+    context.addLifecycleEventListener(this);
+  }
+
+  public void keyboardChanged(Boolean info) {
+    Log.i("Keyboard was changed", String.valueOf(info));
+  }
+
+  public void setKeyboardFocus(int tag) {
+    Activity activity = context.getCurrentActivity();
```

Comment:
final 

compile error: 'getCurrentActivity()' is not public in 'com.facebook.react.bridge.ReactContext'. Cannot be accessed from outside package